fn.go

```diff
@@ -152,7 +152,7 @@ func (x Fn) Marshal(w io.Writer, m int) (int, error) {
 // Unmarshal implements the surge.Unmarshaler interface.
 func (x *Fn) Unmarshal(r io.Reader, m int) (int, error) {
```

Comment:
Similar to above, the signature should be `Unmarshal(but []byte, rem int) ([]byte, int, error)`.

Reply:
See response above.